File: Models/Transaction.cs
namespace Models;

public class Transaction {
    public int id { get; set; }
    public int bankAcctId { get; set; }
    public double amount { get; set; }
    public DateTime date { get; set; }
    public string type { get; set; }
    public string description { get; set; }
}

File: NBOBanking/src/main/java/com/NBOBanking/models/Transaction.java
package com.NBOBanking.models;

import jakarta.persistence.*;

import java.util.Date;

@Entity
@Table(name = "Transactions")
public class Transaction {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "transaction_id", nullable = false)
    public Integer transaction_id;

    @Column(name = "bankaccount_id")
    public int bankaccount_id;

    @Column(name = "transaction_type")
    public String transaction_type;

    @Column(name = "transaction_amount")
    public int transaction_amount;

    @Column(name = "transaction_date")
    public Date transaction_date;
}
